brave/brave-core
# Install ipfs as a normal extension

Fix https://github.com/brave/brave-browser/issues/3950
Fix https://github.com/brave/brave-browser/issues/4218

Instead of installing as a component extension, this PR installs ipfs companion using a webstore installer so manifest could be localized and the extension option page could be shown normally under brave://extensions.

Toggle on the switch would prompt to install IPFS companion extension if it was't installed before. If already installed, it will enable the extension.
Toggle off the switch would disable IPFS companion extension (but not uninstall it). To fully remove (uninstall) the IPFS companion extension after it is installed, remove the extension through brave://extensions.

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on

## Test Plan:
1. Visit brave://settings and toggle IPFS companion switch, a prompt as below will be shown if IPFS companion extension is not currently installed.
<img width="735" alt="Screen Shot 2019-04-24 at 8 18 15 AM" src="https://user-images.githubusercontent.com/4730197/56671491-9a6dd800-6669-11e9-8edd-73291ee44170.png">

2. Click "Cancel" in the prompt, the IPFS companion switch should be changed back to off.

3. Toggle it on again, and click "Add extension" in the prompt to install it, IPFS companion extension should now be installed and enabled.

4. Right click IPFS companion switch icon to open options page, it should shown as
<img width="732" alt="Screen Shot 2019-04-24 at 8 25 58 AM" src="https://user-images.githubusercontent.com/4730197/56672074-9c846680-666a-11e9-8645-025ad4550b8b.png">


5. Toggle it off again and visit brave://extensions, IPFS companion extension should be shown disabled.

6. Enable the IPFS companion switch toggle in brave://settings again, IPFS companion extension should be shown enabled in brave://extensions.

7. Remove IPFS companion extension in brave://extensions, IPFS companion extension switch toggle in brave://settings should be shown off.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
